File: week09/week09-6.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        ans=0
        m,n=len(grid),len(grid[0])
        for i in range(m):
            if grid[i][0]==1: ans+=1
            if grid[i][n-1]==1: ans+=1
        for j in range(n):
            if grid[0][j]==1: ans+=1
            if grid[m-1][j]==1: ans+=1

        for i in range(m-1):
            for j in range(n):
                if grid[i][j]+grid[i+1][j]==1: ans+=1
        for i in range(m):
            for j in range(n-1):
                if grid[i][j]+grid[i][j+1]==1: ans+=1
        return ans

File: week09/week09-6.cpp
int islandPerimeter(int** grid, int gridSize, int* gridColSize) {
    int ans=0;
    int m=gridSize,n=gridColSize[0];
    for(int i=0;i<m;i++){
        if(grid[i][0]==1) ans++;
        if(grid[i][n-1]==1) ans++;
    }
    for(int j=0;j<n;j++){
        if(grid[0][j]==1) ans++;
        if(grid[m-1][j]==1) ans++;
    }
    for(int i=0;i<m-1;i++){
        for(int j=0;j<n;j++){
            if(grid[i][j]+grid[i+1][j]==1) ans++;
        }
    }
    for(int i=0;i<m;i++){
        for(int j=0;j<n-1;j++){
            if(grid[i][j]+grid[i][j+1]==1) ans++;
        }
    }
    return ans;
}
